SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialNetDriverRPC.h

```diff
@@ -0,0 +1,241 @@
+ */
+UCLASS()
+class SPATIALGDK_API USpatialNetDriverRPC : public UObject
+{
+	GENERATED_BODY()
+public:
+	// Definition for a "standard queue", that is, all queued RPC for sending could have an accompanying SpanId.
+	// Makes it easier to define a standard callback for when an outgoing RPC is written to the network.
+	using StandardQueue = SpatialGDK::TWrappedRPCQueue<FSpatialGDKSpanId>;
+
+	USpatialNetDriverRPC();
+	virtual void Init(USpatialNetDriver* InNetDriver, const SpatialGDK::FSubView& InActorAuthSubView,
+					  const SpatialGDK::FSubView& InActorNonAuthSubView);
+
+	void AdvanceView();
+	virtual void ProcessReceivedRPCs();
+	virtual void FlushRPCUpdates();
+	void FlushRPCQueue(StandardQueue& Queue);
+	void FlushRPCQueue(Worker_EntityId, StandardQueue& Queue);
+	TArray<FWorkerComponentData> GetRPCComponentsOnEntityCreation(const Worker_EntityId EntityId);
+
+protected:
+	void MakeRingBufferWithACKSender(FName QueueName, ERPCType RPCType, Worker_ComponentSetId AuthoritySet,
+									 TUniquePtr<SpatialGDK::RPCBufferSender>& SenderPtr,
+									 TUniquePtr<SpatialGDK::TRPCQueue<FRPCPayload, FSpatialGDKSpanId>>& QueuePtr);
+
+	void MakeRingBufferWithACKReceiver(FName QueueName, ERPCType RPCType, Worker_ComponentSetId AuthoritySet,
+									   TUniquePtr<SpatialGDK::TRPCBufferReceiver<FRPCPayload, TimestampAndETWrapper>>& ReceiverPtr);
+
+	virtual void GetRPCComponentsOnEntityCreation(const Worker_EntityId EntityId, TArray<FWorkerComponentData>& OutData);
+
+	struct UpdateToSend
+	{
+		Worker_EntityId EntityId;
+		FWorkerComponentUpdate Update;
+		TArray<FSpatialGDKSpanId> Spans;
+	};
+
+	StandardQueue::SentRPCCallback MakeRPCSentCallback(TArray<UpdateToSend>& OutUpdates);
+	SpatialGDK::RPCCallbacks::DataWritten MakeDataWriteCallback(TArray<FWorkerComponentData>& OutArray) const;
+	SpatialGDK::RPCCallbacks::UpdateWritten MakeUpdateWriteCallback(TArray<UpdateToSend>& OutUpdates) const;
+
+	static void OnRPCSent(SpatialGDK::SpatialEventTracer* EventTracer, TArray<UpdateToSend>& OutUpdates, FName Name,
+						  Worker_EntityId EntityId, Worker_ComponentId ComponentId, uint64 RPCId, const FSpatialGDKSpanId& SpanId);
+	static void OnDataWritten(TArray<FWorkerComponentData>& OutArray, Worker_EntityId EntityId, Worker_ComponentId ComponentId,
+							  Schema_ComponentData* InData);
+	static void OnUpdateWritten(TArray<UpdateToSend>& OutUpdates, Worker_EntityId EntityId, Worker_ComponentId ComponentId,
+								Schema_ComponentUpdate* InUpdate);
+
+	void FlushUpdates(TArray<UpdateToSend>&);
+	bool CanExtractRPC(Worker_EntityId);
+	bool CanExtractRPCOnServer(Worker_EntityId);
+	bool ApplyRPC(Worker_EntityId, SpatialGDK::ReceivedRPC, const FRPCMetaData& MetaData);
+
+	USpatialNetDriver* NetDriver;
+
+	USpatialLatencyTracer* LatencyTracer;
+	SpatialGDK::SpatialEventTracer* EventTracer;
+
+	TUniquePtr<SpatialGDK::RPCService> RPCService;
+	// TUniquePtr<SpatialGDK::TRPCBufferReceiver<FRPCPayload, TimestampAndETWrapper>> NetMulticastReceiver;
```

Comment:
Remove?

Reply:
TODO UNR-5038

Reply:
Add in comment?

Reply:
Or better just remove for now, and add back in 5038